src/project-manager.ts

```diff
@@ -238,6 +238,11 @@ export class ProjectConfiguration {
      */
     private expectedFilePaths = new Set<string>()
 
+    /**
+     * List of uris for all files currently open in the editor
+     */
+    public openFiles: Set<string> = new Set<string>()
```

Comment:
This field is public, but later on you add encapsulation methods `getOpenFiles`, `addToOpenFiles`, `deleteFromOpenFiles`. I would suggest skipping the encapsulation methods using the field directly: `openFiles`, `openFiles.add`, `openFiles.delete`.

Also, can you make `openFiles` a `readonly` field?